sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/adapter/NodeAdapter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.ruby.converter.adapter;
+
+import java.util.Objects;
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyFixnum;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.javasupport.JavaEmbedUtils;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.sonarsource.slang.api.Tree;
+
+@JRubyClass(name = "NodeAdapter")
+public class NodeAdapter extends RubyObject {
```

Comment:
why it doesn't extend `JRubyObject` ?

Reply:
Do you mean `IRubyObject`? `IRubyObject` is an interface, and we don't want to implement all base methods.
`RubyObject` is the equivalent of the ruby `Object` class, so classes that will be used inside the ruby runtime should extend this basic class.